olive/model/hf_utils.py

```diff
@@ -25,6 +27,137 @@ class HFComponent(ConfigBase):
     dummy_inputs_func: Union[str, Callable]
 
 
+class HFModelLoadingArgs(ConfigBase):
+    """
+    Arguments to pass to the `from_pretrained` method of the model class.
+
+    Refer to https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L2074
+    """
+
+    # load model under a specific torch dtype
+    torch_dtype: str = None
+    # A map that specifies where each submodule should go. Refer to link in class docstring
+    # str suffices for torch.dtype, otherwise serializer won't recognize it
+    device_map: Union[int, str, Dict[str, Union[int, str]]] = None
+    # A dictionary device identifier to maximum memory
+    max_memory: Dict = None
+    # quant method to use (transformers.QuantizationMethod), e.g. "bitsandbytes", "gptq"
+    # must be provided if quantization_config is provided
+    quantization_method: str = None
+    # A dictionary of configuration parameters for quantization
+    quantization_config: Dict = None
+    # other kwargs
+    extra_args: Dict = None
```

Comment:
>extra_args [](http://example.com/codeflow?start=4&length=10)

what kinds of parameters that users need to know to be in extra_args or in named args in this class?

Reply:
There are many args here https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L2074

Reply:
What I mean is: should we add document about what kinds of args users need set by extra_args or by others like quantization_method? Since they seems not be consistent. 

For all the args defined in HFModelLoadingArgs, user can specify them directly. But for all other args, user need to set by using extra_args

Reply:
Updated to use pydantic.Field with description